src/main.go

```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	. "screepsgo/src/screeps"
+	"strconv"
+	"syscall/js"
+)
+
+//export Loop
+func Loop() {
+	var spawn = Game.Spawns()["Spawn1"]
+	var room = spawn.Room()
+
+	var myStructures = room.Find(FIND_MY_STRUCTURES)
+	var tower *StructureTower = nil
+	for _, v := range myStructures {

```

Comment:
Room.Find normally returns different types, depending on FIND_ constant. I made it so it only returns RoomObjects, that later need to be converted to their proper type. In case of this loop there is a `var structure = Structure{RoomObject: v}` one line below and from this point on properly typed `structure` is used instead of `v`.